stan/math/prim/scal/prob/cauchy_rng.hpp

```diff
@@ -1,50 +1,60 @@
 #ifndef STAN_MATH_PRIM_SCAL_PROB_CAUCHY_RNG_HPP
 #define STAN_MATH_PRIM_SCAL_PROB_CAUCHY_RNG_HPP
 
-#include <boost/random/cauchy_distribution.hpp>
-#include <boost/random/variate_generator.hpp>
 #include <stan/math/prim/scal/err/check_consistent_sizes.hpp>
 #include <stan/math/prim/scal/err/check_finite.hpp>
-#include <stan/math/prim/scal/err/check_not_nan.hpp>
 #include <stan/math/prim/scal/err/check_positive_finite.hpp>
-#include <stan/math/prim/scal/fun/constants.hpp>
-#include <stan/math/prim/scal/fun/log1p.hpp>
-#include <stan/math/prim/scal/fun/square.hpp>
-#include <stan/math/prim/scal/fun/value_of.hpp>
-#include <stan/math/prim/scal/meta/include_summand.hpp>
+#include <stan/math/prim/scal/meta/max_size.hpp>
+#include <stan/math/prim/scal/meta/scalar_seq_view.hpp>
+#include <stan/math/prim/scal/meta/VectorBuilder.hpp>
+#include <boost/random/cauchy_distribution.hpp>
+#include <boost/random/variate_generator.hpp>
 
 namespace stan {
   namespace math {
-
     /**
      * Return a pseudorandom Cauchy variate for the given location and scale
      * using the specified random number generator.
      *
+     * mu and sigma can each be a scalar, a std::vector, an Eigen::Vector, or
+     * an Eigen::RowVector. Any non-scalar inputs must be the same length.
+     *
+     * @tparam T_loc Type of location parameter
+     * @tparam T_scale Type of scale parameter
      * @tparam RNG type of random number generator
-     * @param mu location parameter
-     * @param sigma positive scale parameter
+     * @param mu (Sequence of) location parameter(s)
+     * @param sigma (Sequence of) scale parameter(s)
      * @param rng random number generator
      * @return Cauchy random variate
      * @throw std::domain_error if mu is infinite or sigma is nonpositive
+     * @throw std::invalid_argument if non-scalars arguments are of different
+     * lengths
      */
-    template <class RNG>
-    inline double
-    cauchy_rng(double mu,
-               double sigma,
-               RNG& rng) {
+    template <typename T_loc, typename T_scale, class RNG>
+    inline typename VectorBuilder<true, double, T_loc, T_scale>::type
+    cauchy_rng(const T_loc& mu, const T_scale& sigma, RNG& rng) {
       using boost::variate_generator;
       using boost::random::cauchy_distribution;
-
       static const char* function = "cauchy_rng";
 
+      scalar_seq_view<T_loc> mu_vec(mu);
+      scalar_seq_view<T_scale> sigma_vec(sigma);
+      size_t N = max_size(mu, sigma);
+      VectorBuilder<true, double, T_loc, T_scale> output(N);
+
       check_finite(function, "Location parameter", mu);
```

Comment:
Checks should go first before any work.  In general, try to put exceptional/odd/edge-case branches first so that when you're by them, you can scan the main branch of the program together.  Would you mind fixing this on all of these?  Sorry for not catchign that earlier.